cmd/fetcher/main.go

```diff
@@ -280,16 +380,71 @@ func createPluginDir(dir string, previousVersion string, newVersion string) (ret
 		filepath.Join(dir, newVersion),
 		previousVersion,
 		newVersion,
+		latestBaseImageVersions,
 	)
 }
 
-func copyFile(src, dest string, prevVersion, newVersion string) error {
-	data, err := os.ReadFile(src)
+func copyFile(
+	src string,
+	dest string,
+	prevVersion string,
+	newVersion string,
+	latestBaseImageVersions map[string]string,
+) (retErr error) {
+	srcFile, err := os.Open(src)
 	if err != nil {
 		return err
 	}
-	replaced := strings.ReplaceAll(string(data), strings.TrimPrefix(prevVersion, "v"), strings.TrimPrefix(newVersion, "v"))
-	return os.WriteFile(dest, []byte(replaced), 0644) //nolint:gosec
+	defer func() {
+		retErr = errors.Join(retErr, srcFile.Close())
+	}()
+	destFile, err := os.OpenFile(dest, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644) //nolint:gosec
+	if err != nil {
+		return err
+	}
+	defer func() {
+		retErr = errors.Join(retErr, destFile.Close())
+	}()
+	filename := filepath.Base(dest)
+	switch filename {
+	case "Dockerfile", "Dockerfile.wasm", "buf.plugin.yaml", "package.json":
```

Comment:
Updated this to be more prescriptive about the places we want to do a global search and replace. The `Dockerfile.wasm` is here as a future placeholder for some work @twoism and I will be doing shortly.